src/libraries/Directory.Build.targets

```diff
@@ -299,4 +299,49 @@
     </PropertyGroup>
   </Target>
 
+  <PropertyGroup>
+    <BeforePack>IncludeAnalyzersInPackage;$(BeforePack)</BeforePack>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <!-- Ensure AnalyzerReference items are restored and built 
+         The target framework of Analyzers has no relationship to that of the refrencing project, 
+         so we don't apply TargetFramework filters nor do we pass in TargetFramework.
+         When BuildProjectReferences=false we make sure to set BuildReference=false to make
+         sure not to try to call GetTargetPath in the outerbuild of the analyzer project. -->
+    <ProjectReference Include="@(AnalyzerReference)" SkipGetTargetFrameworkProperties="true" UndefineProperties="TargetFramework" ReferenceOutputAssembly="false" PrivateAssets="all" BuildReference="$(BuildProjectReferences)"  />
+  </ItemGroup>
+
+  <Target Name="IncludeAnalyzersInPackage" Condition="'@(AnalyzerReference)' != ''">
+    <!-- Call a target in the analyzer project to get all the files it would normally place in a package.
+         These will be returned as items with identity pointing to the built file, and PackagePath metadata
+         set to their location in the package.  IsSymbol metadata will be set to distinguish symbols. -->
+    <MSBuild Projects="@(AnalyzerReference)"
+             Targets="GetAnalyzerPackFiles">
+      <Output TaskParameter="TargetOutputs" ItemName="_AnalyzerFiles" />
+    </MSBuild>
+
+    <ItemGroup>
+      <Content Include="@(_AnalyzerFiles)" Pack="True" Condition="!%(_AnalyzerFiles.IsSymbol)" />
+      <!-- Symbols don't honor PackagePath.  By default they are placed in lib/%(TargetFramework).           
```

Comment:
Is this expected or should we open an issue for it?

Reply:
I believe @Anipik has this on his list.  He hit similar problems with the runtime-specific builds and I used this workaround based on his prototyping.  @Anipik do you have a bug filed on this?

Reply:
yes, i hit the same thing but no we dont have any issue filed for it.

Reply:
Can someone please open an issue for it?

Reply:
I'll file it.  I think it's probably up for debate if they want to allow it.  Regardless I'd want to see tests added to protect the workaround we're currently using.

Reply:
https://github.com/NuGet/NuGet.Client/pull/3980